src/implementations/vanilla/components/navigation-list/_navigation-list.scss

```diff
@@ -71,11 +69,29 @@ $_border: null !default;
   box-shadow: none;
 
   .ecl-navigation-list__image {
-    border: 1px solid $_image-border-color;
+    border: 1px solid map.get($navigation-list, 'image-border-color');
   }
 
   .ecl-navigation-list__content-block {
     padding-inline-start: 0;
     padding-inline-end: 0;
   }
 }
+
+@if map-has-key($navigation-list, 'yellow-bar') {
```

Comment:
nicely done, we should apply this logic to other components too (card for instance), and remove custom rules in theme files